### PR TITLE
net-misc/anydesk: add 6.2.1

### PR DESCRIPTION
**Pull request by jkroonza**
NOTE:  gtkglext dep is still required.
I have however managed to install without dev-libs/atk and run the application and connect to a remote host, I thus believe it's safe to drop atk as dependency.

Closes: https://bugs.gentoo.org/887809
Signed-off-by: Jaco Kroon <jaco@uls.co.za>

### CODE REVIEW

**Review by ceamac — CHANGES_REQUESTED** [review]
Please add `app-accessibility/at-spi2-core:2` to RDEPEND (replaces atk).  Also `media-libs/libglvnd` seems required.

I created bug [#889378](https://bugs.gentoo.org/889378), please add it as a comment to the RDEPEND block with the `x11-libs/gtkglext` situation.

**Comment by jkroonza** [comment]
> `app-accessibility/at-spi2-core:2`

Thank you.  Done.

**Comment by ceamac** [comment]
Still need to add `media-libs/libglvnd[X]` to RDEPEND, it provides `libGL.so.1` and anydesk links against it.

**Comment by jkroonza** [comment]
> Still need to add `media-libs/libglvnd[X]` to RDEPEND, it provides `libGL.so.1` and anydesk links against it.

How did you find this btw?

**Comment by ceamac** [comment]
`qa-vdb` from `app-portage/iwdevtools` is very good at finding missing runtime dependencies.  It reports `media-libs/libglvnd`.  Then I knew from some of my packages that it provides libGL.so.1 only with `[X]` and I looked with `lddtree` to see if this is the case for `anydesk`.

File: net-misc/anydesk/anydesk-6.2.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="8"
+
+inherit desktop optfeature systemd xdg-utils
+
+DESCRIPTION="Feature rich multi-platform remote desktop application"
+HOMEPAGE="https://anydesk.com"
+SRC_URI="https://download.anydesk.com/linux/${P}-amd64.tar.gz
+	https://download.anydesk.com/linux/generic-linux/${P}-amd64.tar.gz"
```
Comment (ceamac) [inline net-misc/anydesk/anydesk-6.2.1.ebuild:11]: Why are there 2 links here with the same name?  The second one returns 404.  Is this some peculiarity about the way Anydesk release their Linux client?

**Review by ceamac — APPROVED** [review]

File: net-misc/anydesk/anydesk-6.2.1.ebuild (hunk 1 above)
Comment (jkroonza) [reply net-misc/anydesk/anydesk-6.2.1.ebuild:11]: Hmm, good spot.

At some point they changed the way the structured their download repository such that old stuff was moved into the generic folder.  Or at least a significant portion of the files.  The folder still exist, and at the time it looked like one of the two URLs should keep working.  I'm OK with only listing the first, it seems they keep at least from version 6.0.1 onwards still there.

I just re-looked at the x86 situation as well, whilst there are no .tar.gz builds available, it does look like there are rpm builds available, I have no interest in that, but if someone needs this on x86 still (I consider this to be unlikely) then we can certainly look at that.

This double download URL was first included in version 5.5.0 which I added due to the move of 5.5.0 into the generic folder, newest versions are not present in the generic folder, meaning that over time the URL changes (in a predictable manner).

**Comment by jkroonza** [comment]
Thank you.

**Pull request closed without merging**